src/dialogs/screenshot/screenshot.vala

```diff
@@ -0,0 +1,1322 @@
+		public bool showtooltips {get; set; default = true;}
+		public bool emit_capture_sound { get; set; default = true; }
+		public int buttonpos {get; private set;}
+		public string tempfile_path {get; private set;}
+		public bool startedfromgui {get; private set; default = false;}
+
+		public signal void buttonpos_changed();
+
+		public CurrentState() {
+			screenshot_settings = new GLib.Settings("org.buddiesofbudgie.budgie-desktop.screenshot");
+
+			buttonplacement = new GLib.Settings("com.solus-project.budgie-wm");
+
+			buttonplacement.changed["button-style"].connect(() => {
+				fill_buttonpos();
+				buttonpos_changed();
+			});
+			fill_buttonpos();
+
+			emit_capture_sound = screenshot_settings.get_boolean("screenshot-capture-sound");
+			screenshot_settings.changed["screenshot-capture-sound"].connect(() => {
+				emit_capture_sound = screenshot_settings.get_boolean("screenshot-capture-sound");
+			});
+
+			showtooltips = screenshot_settings.get_boolean("showtooltips");
+			screenshot_settings.changed["showtooltips"].connect(() => {
+				showtooltips = screenshot_settings.get_boolean("showtooltips");
+			});
+
+			// we need to use the same temporary user-space file across dbus client/server calls to coordinate
+			// the passing of screenshot images
+			string tmpdir = Environment.get_variable("XDG_RUNTIME_DIR") ?? Environment.get_variable("HOME");
+			tempfile_path = GLib.Path.build_path(GLib.Path.DIR_SEPARATOR_S, tmpdir, ".budgiescreenshot_tempfile");
+		}
+
+		private void fill_buttonpos() {
+			buttonpos = (buttonplacement.get_string("button-style") =="left") ? ButtonPlacement.LEFT : ButtonPlacement.RIGHT;
+		}
+
+		public void statechanged(int n) {
+			newstate = n;
+
+			(newstate == 0) ? startedfromgui = false : startedfromgui;
+			(newstate == 1) ? startedfromgui = true : startedfromgui;
+		}
+	}
+
+	[DBus (name = "org.buddiesofbudgie.BudgieScreenshotControl")]
+	public class ScreenshotServer : GLib.Object {
+		CurrentState windowstate;
+
+		public ScreenshotServer() {
+			windowstate = new CurrentState();
+		}
+
+		public async void StartMainWindow() throws Error {
+			if (getcurrentstate() == 0) {
+				new ScreenshotHomeWindow();
+			}
+		}
+
+		public async void StartAreaSelect() throws Error {
+			if (getcurrentstate() == 0) {
+				set_target("Selection");
+				new SelectLayer();
+			}
+		}
+
+		public async void StartWindowScreenshot() throws Error {
+			if (getcurrentstate() == 0) {
+				set_target("Window");
+				new MakeScreenshot(null);
+			}
+		}
+
+		public async void StartFullScreenshot() throws Error {
+			if (getcurrentstate() == 0) {
+				set_target("Screen");
+				new MakeScreenshot(null);
+			}
+		}
+
+		// setup dbus
+		void on_bus_acquired(DBusConnection conn) {
+			// register the bus
+			try {
+				conn.register_object("/org/buddiesofbudgie/ScreenshotControl",	new ScreenshotServer());
+			} catch (IOError e) {
+				warning("on_bus_acquired Could not register service\n");
```

Comment:
I think `warning()` automatically appends a newline to the end of input passed into it.

Reply:
eagle eyed